### PR TITLE
Added throttle util and filter

### PR DESCRIPTION
**Pull request by karevn**
Hello! I think this small addition will be helpful for some cases (at least it was really helpful for my current project).

### CODE REVIEW

**Comment by yyx990803** [comment]
Thanks for the PR - can you give some more information on the actual use case? e.g. what type of UI are you using this for?

**Comment by karevn** [comment]
I am building a "custom cursor" thingie, which should only be displayed when hovering over specific areas of the app. So, I need to check if I should display a custom cursor or default one based on `jQuery(el).closest('bla-blah').size()` call result, which is expensive. But, I don't need an immediate reaction, so if I throttle the event to be only fired once in 100msec - all works fine.

So basically, this thing will be useful for working with mouseover / scroll events when they should be fired "in-between" the action.

File: src/util/lang.js [hunk 1]
```diff
@@ -255,6 +255,47 @@ exports.debounce = function (func, wait) {
...
 
 /**
+* Throttle a function so it will only gets called at most
+* once during a given timespan.
+*
+* @param {Function} func
+* @param {Number} wait
+* @param {Object} options
+* @return {Function} - throttled function
+*/
```
Comment (rafaell-lycan) [inline src/util/lang.js]: :clap:

**Comment by Codebryo** [comment]
is this still open/valid?

**Comment by karevn** [comment]
It's just sitting and waiting :)

**Comment by yyx990803** [comment]
Hey, sorry for letting this sit for so long. The reason I hesitated and decided not to merging this: it's adding weight to the library for a very edge use case, which can be easily implemented as a userland thing. In comparison, `debounce` is in core because it is commonly used on input with `v-model` and there's no easy way for a user to implement that for `v-model`.

Again, thanks for the PR.

**Pull request closed without merging**